### PR TITLE
Fix/1316 CSS prop 

### PR DESCRIPTION
**Pull request by pancaspe87**
This PR addresses a variety of compilation errors - as described in https://github.com/atlassian-labs/compiled/issues/1316 

Types will be addressed in https://github.com/atlassian-labs/compiled/issues/1323

### CODE REVIEW

File: packages/react/types/css.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import type { CssFunction } from '@compiled/react';
+
+declare module 'react' {
```
Comment (pancaspe87) [inline packages/react/types/css.d.ts:3]: To address error such as 
`Property 'css' does not exist on type 'DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>`

Not sure if `CssFunction` is the most appropriate type, let me know if there is better one!

File: packages/babel-plugin/src/utils/traverse-expression/traverse-call-expression.ts [hunk 2]
```diff
@@ -50,6 +50,10 @@ export const traverseCallExpression = (
       // Right now we are only supported these 2 flavors. If we have complex case like `func('arg').fn().variable`,
       // it will not get evaluated.
       if (t.isIdentifier(callee)) {
+        if (callee.name === 'css' && t.isObjectExpression(expression.arguments[0])) {
+          return evaluateExpression(expression.arguments[0], updatedMeta);
+        }
+
         const resolvedBinding = resolveBinding(callee.name, updatedMeta, evaluateExpression);
```
Comment (pancaspe87) [inline packages/babel-plugin/src/utils/traverse-expression/traverse-call-expression.ts:57]: We have a util to check if [isCompiledCSSCallExpression]( https://github.com/atlassian-labs/compiled/blob/87efa985685720886da35766535efc68436f2cb6/packages/babel-plugin/src/utils/is-compiled.ts#L12), but it won't work with imported mixins as it checks for `css` within the given file

File: packages/babel-plugin/src/utils/traverse-expression/traverse-call-expression.ts [hunk 3]
```diff
@@ -50,6 +50,10 @@ export const traverseCallExpression = (
       // Right now we are only supported these 2 flavors. If we have complex case like `func('arg').fn().variable`,
       // it will not get evaluated.
       if (t.isIdentifier(callee)) {
+        if (callee.name === 'css' && t.isObjectExpression(expression.arguments[0])) {
```
Comment (liamqma) [inline packages/babel-plugin/src/utils/traverse-expression/traverse-call-expression.ts:53]: css might be aliased?
```
import { css as css2 } from "@compiled/react"

css2(...)
```
Also, css might have multiple arguments?
```
css({
   color: 'red'
}, {
   background: 'red'
})
```

Comment (pancaspe87) [reply packages/babel-plugin/src/utils/traverse-expression/traverse-call-expression.ts:53]: Both very good points! 

- [ ] css might be aliased
- [ ] css might have multiple arguments

File: packages/react/types/css.d.ts (hunk 1 above)
Comment (JakeLane) [reply packages/react/types/css.d.ts:3]: We need to think about how we're going do this, as this will break jsx runtime overrides

Comment (pancaspe87) [reply packages/react/types/css.d.ts]: Raised https://github.com/atlassian-labs/compiled/issues/1323

**Review by JakeLane — APPROVED** [review]

**Review by liamqma — APPROVED** [review]

File: packages/babel-plugin/src/__tests__/module-traversal.test.ts [hunk 4]
```diff
@@ -589,5 +589,50 @@ describe('module traversal', () => {
 
       expect(actual).toInclude('{color:red}');
     });
+
+    it('should resolve member expression in CSS prop', () => {
```
[the remark this answers is not in the document]
Comment (pancaspe87) [reply packages/babel-plugin/src/__tests__/module-traversal.test.ts:593]: discussed offline. We didn't find any actual usages in products and we think that declarations should resemble CSS and be something like

css({
    color: 'red',
    background: 'red'
  })
  
  Array expressions are supported.

**Pull request merged**